src/smtml/expr.ml

```diff
@@ -501,6 +516,26 @@ let cvtop ty op hte =
   | Ty.Cvtop.String_to_re, _ -> raw_cvtop ty op hte
   | _, Val v -> value (Eval.cvtop ty op v)
   | String_to_float, Cvtop (Ty_real, ToString, real) -> real
+  | Reinterpret_float, Cvtop (Ty_real, Reinterpret_int, e1) -> e1
+  | Reinterpret_float, Cvtop (Ty_fp 32, Reinterpret_int, e1) -> e1
+  | Reinterpret_float, Cvtop (Ty_fp 64, Reinterpret_int, e1) -> e1
```

Comment:
I think the size of e1 should be checked in the two previous examples? And also in the following ones

Reply:
You think so? I was actually just going to suggest something like:

```suggestion
  | Reinterpret_float, Cvtop (Ty_fp n, Reinterpret_int, e) -> 
    assert (match ty with Ty_bitv m -> n = m | _ -> false);
    e
```
I think it would be enough and we could reduce the two pattern matches on `Ty_fp` into a single one

Reply:
Yes, looks good too